tests/e2e/registration-by-sms.js

```diff
@@ -245,7 +245,7 @@ describe('registration transition', () => {
       jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
     });
 
-    const start = moment().startOf('day');
+    const start = moment.utc().startOf('day');
```

Comment:
I'm not sure if that will fix it. If I understand correctly this will change the timezone of the moment instance to UTC. However I think the travis machines run in GMT which is the same so I don't think this will change anything on travis. It will change the timezone when running on dev machines, but I don't think timezones are the issue.

I think what's going wrong is this code is run just before midnight on day `x`, then the assertion is run just after midnight on day `x+1` so the actual text displayed on the page is one day later than expected.

This may seem really unlucky given that there's not much code between this line and the assertion, but protractor is weird and executes a lot of stuff up front before it actually asks the browser to do anything so I suspect this is being calculated at the beginning of the e2e run and then only used much later, which gives a window of several minutes for the day to tick over.

I think the ideal solution would be to fetch the newly generated doc from the server, find the reported_date and add 40 weeks (or whatever) to that. Alternatively I think you could scrape the reported_date from the UI. You could also check that the reported_date is greater than `start` and less than `moment()` to make sure it's generating a date that's accurate enough.

Also note that the next line (249) is useless - moment returns clones of dates when you subtract, so we're cloning it then throwing it away. Either delete the line or append it to the line above (248).